cli/asc.js

```diff
@@ -653,12 +653,16 @@ exports.main = function main(argv, options, callback) {
   // Validate the module if requested
   if (!args.noValidate) {
     stats.validateCount++;
+    let callbackErrorResponse;
```

Comment:
Good catch! Perhaps let's make this an `isValid` var and throw the error `if (!isValid)`?

Reply:
That sounds good to me! :smile:

Reply:
Oh wait, if I'm not mistaken we expect the callback to return a status code? 

So it would be something like:

```
// Validate the module if requested
  if (!args.noValidate) {
    stats.validateCount++;
    let isValid;
    stats.validateTime += measure(() => {
      if (!module.validate()) {
        module.dispose();
        isValid = callback(Error("Validate error")) === 0;
      }
    });
    if (!isValid) {
      return 1;
    }
  }
```

Reply:
Something like the following maybe

```ts
if (!args.noValidate) {
  stats.validateCount++;
  let isValid;
  stats.validateTime += measure(() => {
    isValid = module.validate();
  });
  if (!isValid) {
    module.dispose();
    return callback(Error("validate error"));
  }
}
```

Reply:
Callback there is typically [the default](https://github.com/AssemblyScript/assemblyscript/blob/1d86695dcd27777271ff2ea25dbbc6b7892ce327/cli/asc.js#L220), unless one uses the CLI API. Returns `1` when an error is provided. The returned value becomes the exit code of asc.